hail/src/main/scala/is/hail/expr/ir/analyses/SemanticHash.scala

```diff
@@ -39,11 +41,23 @@ case object SemanticHash extends Logging {
           // Include an encoding of a node's position in the parent's child array
           // to differentiate between IR trees that look identical when flattened
           for ((ir, index) <- levelOrder(nameNormalizedIR)) {
-            encode(ctx.fs, ir, index) match {
-              case Right(bytes) =>
-                hash = extend(hash, bytes)
-              case Left(failure) =>
-                log.warn(s"Failed to compute SemanticHash: $failure")
+            try {
+              val bytes = encode(ctx.fs, ir, index)
+              hash = extend(hash, bytes)
+            } catch {
+              case error@(_: UnsupportedOperationException | _: FileNotFoundException) =>
+                log.info(error)
+                return None
+
+              case NonFatal(error) =>
+                log.warn(
+                  """An internal compiler error occurred.
+                    |Please report this to the Hail Team using the link below,
+                    |including the stack trace at the end of this message.
+                    |https://github.com/hail-is/hail/issues/new/choose
+                    |""".stripMargin,
+                  error
```

Comment:
Seems right that a failure in semhash shouldn't bring the whole query down. It's not that important. Simmer down.
We want to know about these failures though, so a shouty log might be appropriate. Though this needs to be more shouty.